lib/p2p/Pool.ts

```diff
@@ -411,18 +413,50 @@ class Pool extends EventEmitter {
     if (!deal) {
       return;
     }
-    if (!deal.r_preimage) {
-      // TODO: Handle this unexpected case, if we are the maker we should have the preimage.
-      return;
-    }
-    const body: packets.SwapResponsePacketBody = {
-      r_preimage: deal.r_preimage,
-    };
-    const swapResponsePacket = new packets.SwapResponse(body, requestPacket.header.id);
 
-    this.logger.debug('sending back to peer: '  + JSON.stringify(body));
+    let cmdLnd = this.lndLtcClient;
+
+  	switch (deal.makerCurrency) {
+    case 'BTC':
+      break;
+    case 'LTC':
+      cmdLnd = this.lndBtcClient;
+      break;
+  	}
+  	if (!cmdLnd) {
+  	  return;
+  	}
+
+    const request = new lndrpc.SendRequest();
+  	request.setAmt(deal.takerAmount);
+  	request.setDestString(deal.takerPubKey);
+  	request.setPaymentHashString(String(deal.r_hash));
```

Comment:
Rather than using this `String` function to try to cast `string | undefined` into a `string`, I think it's better to do a sanity check up front that `deal.r_hash` is not null.

Reply:
I understand but don't like it so much. The code looks strange when you read it. 

I prefer to remove the undefined option from the definition of deal.r_hash and give it an initial value of `"no hash yet"`. What do you think?

Reply:
Sure, Do it as part of the refactoring....